File: src/main/java/edu/kit/iti/formal/stvs/logic/specification/ConcretizationException.java
package edu.kit.iti.formal.stvs.logic.specification;

/**
 * Created by Philipp on 02.03.2017.
 */
public class ConcretizationException extends Exception {
  private String message;
  private Exception originalException;

  public ConcretizationException(String message) {
    this.message = message;
    originalException = null;
  }

  public ConcretizationException(Exception e) {
    super(e);
    originalException = e;
    message = e.getMessage();
  }

  public String getMessage() {
    return message;
  }

  public Exception getOriginalException() {
    return originalException;
  }
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/logic/specification/ConcretizationException.kt
package edu.kit.iti.formal.stvs.logic.specification

/**
 * Exception that gets thrown if concretization fails in general.
 */
class ConcretizationException : Exception {
    override var message: String
        private set
    var originalException: Exception?
        private set

    constructor(message: String) {
        this.message = message
        originalException = null
    }

    /**
     * Creates an instance by wrapping an existent exception.
     *
     * @param e Exception that causes this exception
     */
    constructor(e: Exception) : super(e) {
        originalException = e
        message = e.message!!
    }
}
